googlemock/test/gmock-matchers_test.cc

```diff
@@ -2708,19 +2708,21 @@ class FloatingPointTest : public testing::Test {
         zero_bits_(Floating(0).bits()),
         one_bits_(Floating(1).bits()),
         infinity_bits_(Floating(Floating::Infinity()).bits()),
-        close_to_positive_zero_(
-            Floating::ReinterpretBits(zero_bits_ + max_ulps_/2)),
-        close_to_negative_zero_(
-            -Floating::ReinterpretBits(zero_bits_ + max_ulps_ - max_ulps_/2)),
+        close_to_positive_zero_(Floating::ReinterpretBits(
+            static_cast<Bits>(zero_bits_ + max_ulps_/2))),
+        close_to_negative_zero_(-Floating::ReinterpretBits(
+            static_cast<Bits>(zero_bits_ + max_ulps_ - max_ulps_/2))),
         further_from_negative_zero_(-Floating::ReinterpretBits(
-            zero_bits_ + max_ulps_ + 1 - max_ulps_/2)),
-        close_to_one_(Floating::ReinterpretBits(one_bits_ + max_ulps_)),
-        further_from_one_(Floating::ReinterpretBits(one_bits_ + max_ulps_ + 1)),
+            static_cast<Bits>(zero_bits_ + max_ulps_ + 1 - max_ulps_/2))),
+        close_to_one_(Floating::ReinterpretBits(
+            static_cast<Bits>(one_bits_ + max_ulps_))),
+        further_from_one_(Floating::ReinterpretBits(
+            static_cast<Bits>(one_bits_ + max_ulps_ + 1))),
         infinity_(Floating::Infinity()),
-        close_to_infinity_(
-            Floating::ReinterpretBits(infinity_bits_ - max_ulps_)),
-        further_from_infinity_(
-            Floating::ReinterpretBits(infinity_bits_ - max_ulps_ - 1)),
+        close_to_infinity_(Floating::ReinterpretBits(
+            static_cast<Bits>(infinity_bits_ - max_ulps_))),
+        further_from_infinity_(Floating::ReinterpretBits(
+            static_cast<Bits>(infinity_bits_ - max_ulps_ - 1))),
```

Comment:
can you write the ReinterpretBits(static_cast<Bits>(...)) as a single short private function to reduce the duplication of ugliness?